ledger/internal/eval_blackbox_test.go

```diff
@@ -422,21 +422,21 @@ ok:
 	return eval, addrs[0], err
 }
 
-// TestEvalAppStateCountsWithTxnGroup ensures txns in a group can't violate app state schema limits
+// TestEvalAppStateCountsWithTxnGroupBlackbox ensures txns in a group can't violate app state schema limits
 // the test ensures that
 // commitToParent -> applyChild copies child's cow state usage counts into parent
 // and the usage counts correctly propagated from parent cow to child cow and back
-func TestEvalAppStateCountsWithTxnGroup(t *testing.T) {
+func TestEvalAppStateCountsWithTxnGroupBlackbox(t *testing.T) {
 	partitiontest.PartitionTest(t)
 
 	_, _, err := testEvalAppGroup(t, basics.StateSchema{NumByteSlice: 1})
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "store bytes count 2 exceeds schema bytes count 1")
 }
 
-// TestEvalAppAllocStateWithTxnGroup ensures roundCowState.deltas and applyStorageDelta
+// TestEvalAppAllocStateWithTxnGroupBlackbox ensures roundCowState.deltas and applyStorageDelta
 // produce correct results when a txn group has storage allocate and storage update actions
-func TestEvalAppAllocStateWithTxnGroup(t *testing.T) {
+func TestEvalAppAllocStateWithTxnGroupBlackbox(t *testing.T) {
```

Comment:
I certainly see nothing wrong with changing these names, but I'm curious if it mattered for some reason.

Reply:
the skipped / multiply-run test counting checker in scripts/buildtools/check_tests.py uses the package + test name from the JUnit file to ensure all tests ran at least once (i.e. no tests were skipped by all shards). It got confused about there being two identically named tests in this package (there is also TestEvalAppStateCountsWithTxnGroup and TestEvalAppAllocStateWithTxnGroup in eval_test.go), so it was reporting it as not having been correctly sharded because the package naming for the blackbox test allows the same test name to be re-used.

Reply:
the JUnit file only reports package and test names, so no ... but I was surprised too. It's because eval_test.go uses "package internal" and eval_blackbox_test.go uses "package internal_test" so the names don't conflict with each other — they're in two different package namespaces

Reply:
Also, the checker's main job is to ensure no test was skipped by all shards, so it is still correct even if this happens in the future.